File: lib/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Napakalaki;

/**
 *
 * @author jaimefrias
 */
public class Monster {

	private String name;
	private int combatLevel;
	private BadConsequence bc;
	private Prize prize;

	Monster(String name, int level, BadConsequence bc, Prize prize) {
		this.name = name;
		combatLevel = level;
		this.bc = bc;
		this.prize = prize;
	}
	// Constructor de copia

	public Monster(Monster monstruo) {
		this.name = monstruo.name;
		this.combatLevel = monstruo.combatLevel;
		this.bc = monstruo.bc;
		this.prize = monstruo.prize;
	}

	String getName() {
		return name;
	}

	int getCombatLevel() {
		return combatLevel;
	}

	BadConsequence getBadConsequence() {
		return bc;
	}

	Prize getPrize() {
		return prize;
	}

	boolean onlyLoseLevels() {
		return bc.onlyLoseLevels();
	}
        
        boolean winLevelsFrom(int lvl) {
                return prize.winLevelsFrom(lvl);
        }
        
        boolean loseTreasure(TreasureKind treasure, boolean visible) {
                return bc.loseTreasure(treasure, visible);
        }

	@Override
	public String toString() {
		return "Name = " + name + " combatLevel = " + Integer.toString(combatLevel);
	}
}

File: lib/monster.rb
require './badconsequence'
require './prize'
class Monster
  
 	#Consultores
	attr_reader :name
	attr_reader :combatLevel
	attr_reader :bc
	attr_reader :prize
  
	#Constructores
	def initialize(name, level, bc, prize)
		@name = name
		@combatLevel = level
		@bc = bc
		@prize = prize
	end
	
	def self.copia(monstruo)
		nuevoObjeto = allocate
		
		@name = monstruo.name
		@combatLevel = monstruo.combatLevel
		@bc = monstruo.bc
		@prize = monstruo.prize
		
		nuevoObjeto #return
	end
	
	def onlyLoseLevels
		bc.onlyLoseLevels
	end
	
	def winLevelsFrom(lvl)
		prize.winLevelsFrom(lvl)
	end
	
	def loseTreasure(treasure, visible)
		bc.loseTreasure(treasure, visible)
	end
	
	def to_s
		"Name: #{@name}\nCombat level: #{@combatLevel}"
	end
	
	def getLevelsGained
		@bc.levels
	end
	
	def getTreasuresGained
		@bc.nVisibleTreasures
	end
	
	#El método new tiene visibilidad privada
	#private_class_method :new
end
